pkg/api/login_oauth.go

```diff
@@ -214,7 +214,15 @@ func (hs *HTTPServer) OAuthLogin(ctx *m.ReqContext) {
 	metrics.MApiLoginOAuth.Inc()
 
 	if redirectTo, _ := url.QueryUnescape(ctx.GetCookie("redirect_to")); len(redirectTo) > 0 {
-		ctx.SetCookie("redirect_to", "", -1, setting.AppSubUrl+"/")
+		ctx.Resp.Header().Del("Set-Cookie")
```

Comment:
You cannot delete the "Set-Cookie" header since `hs.loginUserWithUser` above sets the auth cookie using `http.SetCookie`. If you delete the "Set-Cookie" header here you basically logging out the user.

Reply:
Make sense, I've deleted the 'Set-Cookie' delete in the code. Thanks for noticing.